src/Microsoft.SqlTools.ServiceLayer/QueryExecution/ResultSet.cs

```diff
@@ -313,14 +327,20 @@ public Task<ExecutionPlan> GetExecutionPlan()
         /// <summary>
         /// Reads from the reader until there are no more results to read
         /// </summary>
+        /// <param name="dbDataReader">The data reader for getting results from the db</param>
         /// <param name="cancellationToken">Cancellation token for cancelling the query</param>
-        public async Task ReadResultToEnd(CancellationToken cancellationToken)
+        public async Task ReadResultToEnd(DbDataReader dbDataReader, CancellationToken cancellationToken)
         {
+            // Sanity check to make sure we got a reader
+            Validate.IsNotNull(nameof(dbDataReader), SR.QueryServiceResultSetReaderNull);
+
             try
```

Comment:
Although there was a nasty bug here where I was checking if the error string was null...